models/gif_cond_exp.cpp

```diff
@@ -0,0 +1,594 @@
+    if ( tau_sfa_[ i ] <= 0 )
+      throw BadProperty( "All time constants must be strictly positive." );
+
+  for ( size_t i = 0; i < tau_stc_.size(); i++ )
+    if ( tau_stc_[ i ] <= 0 )
+      throw BadProperty( "All time constants must be strictly positive." );
+
+  if ( tau_synE_ <= 0 || tau_synI_ <= 0 )
+    throw BadProperty( "Synapse time constants must be strictly positive." );
+}
+
+void
+nest::gif_cond_exp::State_::get( DictionaryDatum& d,
+  const Parameters_& p ) const
+{
+  def< double >( d, names::V_m, y_[ V_M ] ); // Membrane potential
+  def< double >( d, names::E_sfa, sfa_ );    // Adaptive threshold potential
+  def< double >( d, names::stc, stc_ );      // Spike-triggered current
+}
+
+void
+nest::gif_cond_exp::State_::set( const DictionaryDatum& d,
+  const Parameters_& p )
+{
+  updateValue< double >( d, names::V_m, y_[ V_M ] );
+}
+
+nest::gif_cond_exp::Buffers_::Buffers_( gif_cond_exp& n )
+  : logger_( n )
+  , s_( 0 )
+  , c_( 0 )
+  , e_( 0 )
+{
+  // Initialization of the remaining members is deferred to
+  // init_buffers_().
+}
+
+nest::gif_cond_exp::Buffers_::Buffers_( const Buffers_&, gif_cond_exp& n )
+  : logger_( n )
+  , s_( 0 )
+  , c_( 0 )
+  , e_( 0 )
+{
+  // Initialization of the remaining members is deferred to
+  // init_buffers_().
+}
+
+/* ----------------------------------------------------------------
+ * Default and copy constructor for node
+ * ---------------------------------------------------------------- */
+
+nest::gif_cond_exp::gif_cond_exp()
+  : Archiving_Node()
+  , P_()
+  , S_( P_ )
+  , B_( *this )
+{
+  recordablesMap_.create();
+}
+
+nest::gif_cond_exp::gif_cond_exp( const gif_cond_exp& n )
+  : Archiving_Node( n )
+  , P_( n.P_ )
+  , S_( n.S_ )
+  , B_( n.B_, *this )
+{
+}
+
+nest::gif_cond_exp::~gif_cond_exp()
+{
+  // GSL structs may not have been allocated, so we need to protect destruction
+  if ( B_.s_ )
+    gsl_odeiv_step_free( B_.s_ );
+  if ( B_.c_ )
+    gsl_odeiv_control_free( B_.c_ );
+  if ( B_.e_ )
+    gsl_odeiv_evolve_free( B_.e_ );
+}
+
+/* ----------------------------------------------------------------
+ * Node initialization functions
+ * ---------------------------------------------------------------- */
+
+void
+nest::gif_cond_exp::init_state_( const Node& proto )
+{
+  const gif_cond_exp& pr = downcast< gif_cond_exp >( proto );
+  S_ = pr.S_;
+}
+
+void
+nest::gif_cond_exp::init_buffers_()
+{
+  B_.spike_exc_.clear(); // includes resize
+  B_.spike_inh_.clear(); // includes resize
+  B_.currents_.clear();  //!< includes resize
+  B_.logger_.reset();    //!< includes resize
+  Archiving_Node::clear_history();
+
+  B_.step_ = Time::get_resolution().get_ms();
+  B_.IntegrationStep_ = B_.step_;
+
+  if ( B_.s_ == 0 )
+    B_.s_ =
+      gsl_odeiv_step_alloc( gsl_odeiv_step_rkf45, State_::STATE_VEC_SIZE );
+  else
+    gsl_odeiv_step_reset( B_.s_ );
+
+  if ( B_.c_ == 0 )
+    B_.c_ = gsl_odeiv_control_y_new( 1e-3, 0.0 );
```

Comment:
Maybe you want to allow the user to set the error limit via a parameter instead of hard-coding `1e-3`?